### ISSUE TITLE
new rule: no-side-effects-wait-for

### ISSUE THREAD

**Issue by renatoagds** [testing-library/eslint-plugin-testing-library#134]
Following the proposed for Kent at his post (https://kentcdodds.com/blog/common-mistakes-with-react-testing-library). I would like to suggest a new rule that detects side effects inside a wait-for:

```javascript
❌ // wrong
await waitFor(() => {
  fireEvent.click(a)
  expect(b).toEqual('b')
})
```

```javascript
✅ // correct
fireEvent.click(a)
await waitFor(() => expect(b).toEqual('b'))
```

**Comment by Belco90** [comment]
This one duplicates #133

**Comment by Belco90** [comment]
Sorry, I closed this one by mistake! Thanks again Renato, this suggestion sounds good too.

We would have to identify which _side effects_ we want to check here, I guess those from testing library itself.

Again, would like to give it a try? PRs are welcome!

**Comment by renatoagds** [comment]
> We would have to identify which side effects we want to check here, I guess those from testing library itself.

@Belco90 Yes! I was think to add `fireEvent` and `userEvent` initially. Sounds good?

**Comment by Belco90** [comment]
Sounds good!

**Comment by renatoagds** [comment]
Just to update! I'm finishing #133 this week and will start work on this one 🚀

**Comment by Belco90** [comment]
Awesome, thanks for your help 💪
